src/vaults/VaultInternal.ts

```diff
@@ -48,44 +48,45 @@ class VaultInternal {
     fresh?: boolean;
   }) {
     logger = logger ?? new Logger(this.constructor.name);
+    const dir = path.join(vaultsUtils.makeVaultIdPretty(vaultId), 'contents');
+    const gitdir = path.join(vaultsUtils.makeVaultIdPretty(vaultId), '.git');
     if (fresh) {
+      logger.info(
+        `Initialising Vault '${vaultsUtils.makeVaultIdPretty(vaultId)}'`,
+      );
       try {
-        await efs.rmdir(makeVaultIdPretty(vaultId), { recursive: true });
+        await efs.rmdir(vaultsUtils.makeVaultIdPretty(vaultId), {
+          recursive: true,
+        });
       } catch (err) {
         if (err.code !== 'ENOENT') {
           throw err;
         }
       }
-      await efs.mkdir(path.join(makeVaultIdPretty(vaultId), 'contents'), {
+      await efs.mkdir(dir, {
         recursive: true,
       });
-      const efsVault = await efs.chroot(
-        path.join(makeVaultIdPretty(vaultId), 'contents'),
-      );
+      const efsVault = await efs.chroot(dir);
       await efsVault.start();
-      // Creating a new vault.
       await git.init({
         fs: efs,
-        dir: path.join(makeVaultIdPretty(vaultId), 'contents'),
-        gitdir: path.join(makeVaultIdPretty(vaultId), '.git'),
+        dir,
+        gitdir,
       });
       const workingDir = await git.commit({
         fs: efs,
-        dir: path.join(makeVaultIdPretty(vaultId), 'contents'),
-        gitdir: path.join(makeVaultIdPretty(vaultId), '.git'),
+        dir,
+        gitdir,
         author: {
-          name: makeVaultIdPretty(vaultId),
+          name: keyManager.getNodeId(),
         },
         message: 'Initial Commit',
       });
       await efs.writeFile(
```

Comment:
In another PR I might have a look at the need to write this file here. I remember that cloning and pulling with iso-git fails without (and for some reason it isnt automatically written). This may be because we are looking for it in the git utility functions rather than it being a requirement in the iso-git library. See the `packedrefs` function in `git/utils.ts`

Reply:
https://github.com/MatrixAI/js-polykey/issues/298